src/huggingface_hub/serialization/_torch.py

```diff
@@ -644,3 +979,18 @@ def _get_dtype_size(dtype: "torch.dtype") -> int:
         _float8_e5m2: 1,
     }
     return _SIZE[dtype]
+
+
+class _IncompatibleKeys(namedtuple("IncompatibleKeys", ["missing_keys", "unexpected_keys"])):
+    """
+    This is used to report missing and unexpected keys in the state dict.
+    Taken from https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/module.py#L52.
+
+    """
+
+    def __repr__(self) -> str:
+        if not self.missing_keys and not self.unexpected_keys:
+            return "<All keys matched successfully>"
+        return super().__repr__()
+
+    __str__ = __repr__
```

Comment:
I would tend to prefer a dataclass here for consistency with the rest of the library

Reply:
The choice of using a `NamedTuple` for `_IncompatibleKeys` is to maintain a consistency between the output of `_load_sharded_checkpoint()` and `nn.Module.load_state_dict()`. This makes it easier for the integration in transformers (I will have a PR for this shortly). I'd rather have this uniformity to simplify the handling of the return values of `load_torch_model()`.